### PR TITLE
Added trace_filter API implementation

### PR DESCRIPTION
**Pull request by Sixtysixter**
trace_filter API implementation

### CODE REVIEW

File: silkrpc/core/evm_trace.hpp [hunk 1]
```diff
@@ -358,12 +381,15 @@ class TraceCallExecutor {
         return execute(block.header.number-1, block, transaction, transaction.transaction_index, config);
     }
     boost::asio::awaitable<std::vector<Trace>> trace_transaction(const silkworm::BlockWithHash& block, const silkrpc::Transaction& transaction);
+    boost::asio::awaitable<TraceFilterResult> trace_filter(const TraceFilter& trace_filter);
 
 private:
+    // boost::asio::awaitable<std::vector<Trace>> block_traces(const silkworm::BlockWithHash& block_with_hash);
```
Comment (canepat) [inline silkrpc/core/evm_trace.hpp:387]: Please remove commented code

**Review by canepat — CHANGES_REQUESTED** [review]
You should also update the `trace_filter` status in API doc

File: silkrpc/core/evm_trace.cpp [hunk 2]
```diff
@@ -240,6 +289,18 @@ void to_json(nlohmann::json& json, const TraceManyCallResult& result) {
...
+//     json["blockHash"] = nlohmann::json::value_t::null;
+//     json["blockNumber"] = nlohmann::json::value_t::null;
+//     json["result"] = nlohmann::json::value_t::null;
+//     json["subtraces"] = nlohmann::json::value_t::null;
+//     json["traceAddress"] = nlohmann::json::value_t::null;
+//     json["transactionHash"] = nlohmann::json::value_t::null;
+//     json["transactionPosition"] = nlohmann::json::value_t::null;
+//     json["type"] = nlohmann::json::value_t::null;
+// }
+
```
[the remark this answers is not in the document]
Comment (Sixtysixter) [reply silkrpc/core/evm_trace.cpp:303]: Done

[the remark this answers is not in the document]
Comment (Sixtysixter) [reply silkrpc/core/evm_trace.cpp:303]: done

**Review by canepat — APPROVED** [review]

**Pull request merged**